pulsar-broker-common/src/main/java/org/apache/pulsar/bookie/rackawareness/BookieRackAffinityMapping.java

```diff
@@ -133,6 +137,28 @@ public synchronized void setConf(Configuration conf) {
         }
     }
 
+    private void watchAvailableBookies() {
+        BookieAddressResolver bookieAddressResolver = getBookieAddressResolver();
+        if (bookieAddressResolver instanceof DefaultBookieAddressResolver) {
+            try {
+                Field field = DefaultBookieAddressResolver.class.getDeclaredField("registrationClient");
```

Comment:
Can you please send a PR to BookKeeper in order to make this field accessible with a getter?

Reply:
That's fine.

Reply:
https://github.com/apache/bookkeeper/pull/3724.

Is this pulsar pr waiting for the next bk release?

Reply:
We can use reflection